File: src/back/src/main/java/br/gov/ce/saude/tsp/domains/entities/ObjetivosEstrategicos.java
package br.gov.ce.saude.tsp.domains.entities;

import br.gov.ce.saude.tsp.common.Auditoria;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "objetivos_estrategicos")
public class ObjetivosEstrategicos extends Auditoria {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "secretaria_executiva")
    private String secretariaExecutiva;

    @Column(name = "coordenadoria")
    private String coordenadoria;

    @Column(name = "descricao_objetivo")
    private String descricaoObjetivo;
}

File: front/src/app/core/models/objetivos-estrategicos.ts
export class ObjetivosEstrategicos {
  public id?: number;
  public secretariaExecutiva!: string;
  public coordenadoria!: string;
  public descricaoObjetivo?: string;
  public ativo?: boolean;
  
  constructor(init?: Partial<ObjetivosEstrategicos>) {
    Object.assign(this, init);
  }
}
